utils/writeable/src/lib.rs

```diff
@@ -71,12 +71,22 @@ mod cmp;
 mod either;
 mod impls;
 mod ops;
+mod parts_write_adapter;
+mod testing;
+mod try_writeable;
 
 use alloc::borrow::Cow;
 use alloc::string::String;
-use alloc::vec::Vec;
 use core::fmt;
 
+pub use try_writeable::TryWriteable;
+
+#[doc(hidden)]
+pub mod _internal {
+    pub use super::testing::try_writeable_to_parts_for_test;
+    pub use super::testing::writeable_to_parts_for_test;
+}
```

Comment:
```suggestion
pub use testing as _internal;
```
or `as _testing`, or just `pub mod testing`?

Reply:
We have

```rust
#[doc(hidden)]
pub mod _internal {
```

for macros in `icu_provider` and `zerovec`.

Reply:
I guess, however those aren't just testing things.

Reply:
Internal exports for macros are internal exports for macros. Doesn't matter whether they are for testing or not. They are exported across crate boundaries so I would like the fully qualified name to have "internal".

Reply:
Ok but still this can be simplified as `pub use testing as _internal;`

Reply:
No, I actually prefer an explicit `pub mod _internal` with the list of symbols inside. It is more clear what is being exported and more general if we later add more macro functions for some future non-test macro.